src/main/java/org/broadinstitute/hellbender/utils/genotyper/AlleleLikelihoods.java

```diff
@@ -1129,60 +1112,50 @@ private void removeEvidence(final int sampleIndex, final Collection<EVIDENCE> ev
         removeEvidenceByIndex(sampleIndex, indexesToRemove);
     }
 
+    // remove evidence and unset the {@code evidenceIndexBySampleIndex} cache for this sample
     // assumes that evidencesToRemove is sorted and without duplicates.
     private void removeEvidenceByIndex(final int sampleIndex, final int[] evidencesToRemove) {
-        if (evidencesToRemove.length == 0) {
+        final int numToRemove = evidencesToRemove.length;
+        if (numToRemove == 0) {
             return;
-        } else {
-            final Object2IntMap<EVIDENCE> evidenceIndex = evidenceIndexBySampleIndex(sampleIndex);
-            final int oldEvidenceCount = numberOfEvidences[sampleIndex];
-            final int newEvidenceCount = oldEvidenceCount - evidencesToRemove.length;
-            if (newEvidenceCount < 0) {
-                throw new IllegalStateException("attempt to remove non-existent evidence or repeated evidence");
-            } else if (newEvidenceCount == 0) { // taking in consideration the assumption we simply remove
-                // all evidence.
-                evidenceIndex.clear();
-                numberOfEvidences[sampleIndex] = 0;
+        }
+        final int oldEvidenceCount = numberOfEvidences[sampleIndex];
+        final int newEvidenceCount = oldEvidenceCount - evidencesToRemove.length;
+        Utils.validate(newEvidenceCount >= 0, "attempted to remove non-existent evidence or repeated evidence");
```

Comment:
This method is ```private``` do we really need to check this, can we simply assume the input is correct?

Reply:
Works for me.  Done.